### DISCUSSION TITLE
Sandbox- Podfile sync error

### DISCUSSION THREAD

**Discussion by dcavusoglu** [codemagic-ci-cd/codemagic-docs#1783] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

In one of my projects, the app was always crashing, that's why I deleted all views and components, but the splash screen view. I made it work in Xcode and tried to build on Codemagic to test my app on the device. No dependencies in pubspec. Also, after a few tries, I removed ios folder and created a new one and tried with it. 

updated cocoapods. 

did pod deintegrate
pod update 
pod install many times

cleaned build history in Xcode
deleted Location/ Derived Data folder
deleted Posd_Runner.framework under Build Phases/ Link Binary With Libraries

nothing worked

It is working with Xcode and Simulator, but not in codemagic. 

### Expected results

I just expect the app to be built and run on my test device. 

### Actual results

However,  I always get the same error. 

`Updating project for Xcode compatibility.
Upgrading Runner.xcscheme
Automatically signing iOS for device deployment using specified development team in Xcode project: 7M5HM66M5Z
Running Xcode build...                                          
Xcode archive done.                                          2.0s
Failed to build iOS app
Error (Xcode): The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.

Encountered error while archiving for device.

Build failed :|
Failed to build for iOS` 

### Build id (optional)

648c56c37eb121e5c402183

**Comment by himesh-cm** [comment]
Hi @dcavusoglu the build id `648c56c37eb121e5c402183` is not valid. Can you please check and share the entire url?

**Reply by dcavusoglu** [reply]
Hi @himesh-cm thank you, here is the url: https://codemagic.io/app/647da4527610fe9dc12aa0a1/build/648c56c37eb1f21e5c402183

**Reply by himesh-cm** [reply]
Hi @dcavusoglu, I see that the latest build was successful, and you could resolve it on your own.
Can you please confirm if you are still looking for assistance?

**Reply by dcavusoglu** [reply]
hi @himesh-cm, actually the last build was not the solution for us it was just a trial project we created, a blank app. We thought we could make it from scratch but a bunch of other issues occurred. So, the problem with our project which is the topic of this thread is still going on. If you had a chance to check out the build whose link and id we sent, it would be great. Thanks.

**Reply by himesh-cm** [reply]
Hi @dcavusoglu, i see that you are running the app in `debug` mode. Can you specify `release` under `workflow settings > Build >mode` and share a build id?

**Reply by himesh-cm** [reply]
@dcavusoglu, Have you tried the solution mentioned in my previous message?

**Comment by himesh-cm** [comment]
Hi @dcavusoglu, I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.

**Reply by dcavusoglu** [reply]
Thank you Himesh we could not find a solution yet, tried the ways you mentioned before we opened this thread. It was a pending issue for us still bec of other tasks of ours. Thanks for your support.

**Comment by taskindrp** [comment]
Hi @himesh-cm 

Suddenly, I'm getting this error, build id is `650acd98ddb387bb2425b00f`

<img width="885" alt="CleanShot 2023-09-20 at 13 56 53@2x" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/60576788/0f648dfb-863e-4049-b9d3-c7b628957ed7">

**Comment by jitendramistry1** [comment]
@dcavusoglu , I'm getting the same issue, Have you found any solution for the below error:

Error (Xcode): The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.
